api/Node.json

```diff
@@ -1532,20 +1532,22 @@
             },
             "firefox": [
               {
-                "version_added": true
+                "version_added": "9"
               },
               {
-                "version_added": "9",
-                "notes": "The <code>ownerDocument</code> of doctype nodes (that is, nodes for which <code>Node.nodeType</code> is <code>Node.DOCUMENT_TYPE_NODE</code> or 10) is no longer null. Instead, the <code>ownerDocument</code> is the document on which <code>document.implementation.createDocumentType()</code> was called."
+                "version_added": "1",
+                "version_removed": "9",
+                "notes": "Before Firefox 9, the <code>ownerDocument</code> of doctype nodes (that is, nodes for which <code>Node.nodeType</code> is <code>Node.DOCUMENT_TYPE_NODE</code> or 10) is <code>null</code>. Since Firefox 9, the <code>ownerDocument</code> is the document on which <code>document.implementation.createDocumentType()</code> was called."
```

Comment:
Can we avoid mentioning the version number here, instead just saying what the problem was. And should we have `partial_implementation`?

Reply:
I feel that adding the version number into the note helps give it just a little bit more context, personally, but if desired, I can remove the version number from the note.

Reply:
I will have to defer to @ddbeck and @Elchi3 on note style.

Reply:
In terms of style, a note should make sense with respect to the versions in the support statement. This note is weird: the first part applies to the entirety of the support statement, while the second applies the support statement above it.

If this is fairly represented as two support statements, then the note should be split across the two support statements and neither should mention a version number (since they both apply to the full range).

The note text could be preserved as-is, if the two support statements were combined (i.e., `"version_added": "1"`, no `"version_removed"`, and the existing note), since you'd need the version numbers to show applicability for the note.

Reply:
This note is about a single change, but expressed in terms of two ranges. If I'm reading the note correctly, it's that `document.implementation.createDocumentType('html', '', '').ownerDocument` was null in Firefox 1-8, but in Firefox 9 was fixed such that `document.implementation.createDocumentType('html', '', '').ownerDocument === document` holds true.

https://bugzilla.mozilla.org/show_bug.cgi?id=675166 is the fix that led to this note, added here:
https://wiki.developer.mozilla.org/en-US/docs/Web/API/DOMImplementation/createDocumentType$compare?to=66817&from=66816

My inclination would be to just remove the note, but if it is kept, it could be written as `"The <code>ownerDocument</code> of doctype nodes (that is, nodes for which <code>Node.nodeType</code> is <code>Node.DOCUMENT_TYPE_NODE</code> or 10) is <code>null</code> instead of the document on which <code>document.implementation.createDocumentType()</code> was called."`

Reply:
@Elchi3 I see your thumbs up, but am not sure if that subscribes you to the thread, so I'd also like your thoughts on a single note.

The angle I'm coming from mostly with notes is that if everything is fine with recent versions of browsers, I don't want to see a note that I have to expand and understand. Putting it in its own range solve that.

Reply:
hm, I think the note you formulated belongs in the documentation and not in compat data?

I also wonder if https://github.com/mdn/browser-compat-data/blob/master/docs/data-guidelines.md#removal-of-irrelevant-features should apply to notes. I think I'm hearing from you that this note is now irrelevant and I think you're right.

Reply:
I don't understand this API enough to actually answer the question myself, but suppose I naively split the note, like this:

```json
[
  {
    "version_added": "9",
    "notes": "The <code>ownerDocument</code> of doctype nodes (that is, nodes for which <code>Node.nodeType</code> is <code>Node.DOCUMENT_TYPE_NODE</code> or 10) is the document on which <code>document.implementation.createDocumentType()</code> was called."
  },
  {
    "version_added": "1",
    "version_removed": "9",
    "notes": "The <code>ownerDocument</code> of doctype nodes (that is, nodes for which <code>Node.nodeType</code> is <code>Node.DOCUMENT_TYPE_NODE</code> or 10) is <code>null</code>."
  }
]
```

I think what you're saying is that the first note is pointless (that this is specified or at least compatible behavior). If that's the case, then yes, omit the first note (but not the second). What you can't do is stuff a note about the first support statement into the second support statement to make the first support statement look artificially uncomplicated.

Reply:
The first of those notes is indeed pointless, because this is per spec and interoperable. It's tested by https://wpt.fyi/results/dom/nodes/DOMImplementation-createDocumentType.html and that test passes everywhere.

Reply:
Are we okay with this then?

```json
[
  {
    "version_added": "9"
  },
  {
    "version_added": "1",
    "version_removed": "9",
    "notes": "The <code>ownerDocument</code> of doctype nodes (that is, nodes for which <code>Node.nodeType</code> is <code>Node.DOCUMENT_TYPE_NODE</code> or 10) is <code>null</code>."
  }
]
```

I think it would work in this case but in general it also presents problems. I have seen compat data like this:

```json
[
  {
    "version_added": "85"
  },
  {
    "version_added": "1",
    "version_removed": "84",
    "notes": "a minor problem that was present before version 85"
  }
]
```

At first sight it looks like it is a new feature introduced in version 85. However, if you expand and look closer, you realize it is an old feature with great support dating back really far with just a minor fix that happened in version 85.

In cases like this I usually advocate to make it an own sub feature instead of cramping it is array support statements. If all other browsers always had the correct behavior then they get the compat info from the main feature.

Reply:
Yeah, I think we should do the first one:
```json
[
  {
    "version_added": "9"
  },
  {
    "version_added": "1",
    "version_removed": "9",
    "notes": "The <code>ownerDocument</code> of doctype nodes (that is, nodes for which <code>Node.nodeType</code> is <code>Node.DOCUMENT_TYPE_NODE</code> or 10) is <code>null</code>."
  }
]
```

The way this is presented on MDN is unfortunate, and indeed will make it look, at first glance, like the feature is newer than it actually is. This is indeed more of a problem for a more recent change. Notably, the caniuse.com current-release-aligned style doesn't have this issue.

Reply:
I have updated the notes now.

Reply:
Good point, another reason to investigate a better rendering for MDN.